addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/AlertsCacheableScannerUnitTest.java

```diff
@@ -0,0 +1,331 @@
+
+    // Then
+    assertAlertEvidence("no-cache");
+  }
+
+  @Test
+  public void shouldRaiseAlertWhenCacheIsFreshAndS_MaxAgeDirectiveIsSet()
+      throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1");
+    msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Cache-Control: s-maxage=100, public");
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertAlertEvidence("s-maxage=100");
+  }
+
+  @Test
+  public void shouldRaiseAlertWhenCacheIsFreshAndMaxAgeDirectiveIsSet()
+      throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1");
+    msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Cache-Control: max-age=100,public");
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertAlertEvidence("max-age=100,public");
+  }
+
+  @Test
+  public void shouldRaiseAlertWhenCacheIsFreshAndExpirySpecified()
+      throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1");
+    msg.setResponseHeader(
+        "HTTP/1.1 200 OK\r\n"
+            + "Cache-Control: public\r\n"
+            + "Expires: Wed, 02 Oct 2019 08:00:00 GMT\r\n"
+            + "Date: Wed, 02 Oct 2019 07:00:00 GMT");
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertAlertEvidence("Wed, 02 Oct 2019 08:00:00 GMT");
+  }
+
+  @Test
+  public void shouldRaiseAlertCacheIsFreshWhenNoLifetimeSpecified()
+      throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1");
+    msg.setResponseHeader("HTTP/1.1 200 OK\r\n" + "Cache-Control: public");
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertThat(alertsRaised.size(), equalTo(1));
+    assertThat(alertsRaised.get(0).getEvidence().length(), equalTo(0));
+    assertThat(
+        alertsRaised.get(0).getOtherInfo(),
+        equalTo(
+            Constant.messages.getString(
+                "pscanalpha.storablecacheable.otherinfo.liberallifetimeheuristic")));
+  }
+
+  @Test
+  public void shouldRaiseAlertWhenStaleRetrieveAllowed() throws HttpMalformedHeaderException {
+    // Given
+    HttpMessage msg = new HttpMessage();
+    msg.setRequestHeader("GET / HTTP/1.1");
+    msg.setResponseHeader(
+        "HTTP/1.1 200 OK\r\n"
+            + "Cache-Control: public\r\n"
+            + "Expires: Wed, 02 Oct 2019 06:00:00 GMT\r\n"
+            + "Date: Wed, 02 Oct 2019 07:00:00 GMT");
+
+    // When
+    rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+    // Then
+    assertThat(alertsRaised.size(), equalTo(1));
+    assertThat(alertsRaised.get(0).getEvidence(), equalTo(""));
```

Comment:
Isn't this the equivalent of the assertion (`assertThat(alertsRaised.get(0).getEvidence().length(), equalTo(0));`) in the previous test? Staying consistent is likely best. Also maybe `assertTrue(alertsRaised.get(0).getEvidence().isEmpty())` would be better? Thoughts @thc202 / @psiinon?

Reply:
Will be resolved with the next commit.